### ISSUE TITLE
realtionship with native jenkins pipelines (jenkinsfile)

### ISSUE THREAD

**Issue by ssbarnea** [lexandro/integration-bootstrap#1]
I think that while your articles related to DSL Job Plugin are extreamly usueful, I do find that they lack one aspect: how does this related to Jenkins2 pipelines.

I am asking this because I do want to generate J2 pipelines (Jenkinsfile) and somehow I struggle to find a good way to adapt our current use of JJB to generate pipelines instead of classic jobs. While JJB can be used to generate pipelines it seems to do a very poor job at allowing you to compose a pipeline script out of reusable snippets.

I wonder if the DSL Job plugin could help here or not. The trick is that I do really find useful to compile/geberate these pictures pipelines because they are easy to read and to track changes in them (as opposed to the old XML).

**Comment by lexandro** [comment]
You're totally right, but this article has been written _before_ jenkins 2.0 and the Jenkinsfile based delivery pipeline usage.

Btw. DSL plugin creates _static_ structure for your builds, but J2 uses on-the-fly approach.

**Comment by ssbarnea** [comment]
So the big question would be: what is the best way to generate new-pipelines programmately?

**Comment by lexandro** [comment]
Static pipeline, with fix flow: job dsl
Dynamic pipeline: Jenkinsfile+jenkins 2.0 

Pick what you want.

IMHO a third solution could be the best: convention over configuration+ json/yml/xml based build config. The weak point of the first two solution is the groovy based approach. The syntax can change, compatibilty issue can happen. IF you introduce an abstraction layer above those with an extensible configuration format, you can change the underlying engine anytime when you want and you can extend the functionality with your own plugins. It looks like the travis/shippable/drone.io way, but with the power to add your own customizations. That would be awesome :)
